cpp/src/arrow/array/array_nested.cc

```diff
@@ -33,7 +33,6 @@
 #include "arrow/type.h"
 #include "arrow/type_fwd.h"
 #include "arrow/type_traits.h"
-#include "arrow/util/atomic_shared_ptr.h"
```

Comment:
Technically I suppose we should add `#include <atomic>` for IWYU but it is probably transitively included by most standard includes.

Reply:
You're right, let's do it.